appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardContext.java

```diff
@@ -4549,6 +4555,40 @@ protected void callServletContainerInitializers() throws LifecycleException {
         }
     }
 
+    private LinkedList<Entry<Class<? extends ServletContainerInitializer>, Set<Class<?>>>> orderInitializers(Map<Class<? extends ServletContainerInitializer>, Set<Class<?>>> initializerList) {
```

Comment:
Later we could add an option to glassfish-web.xml to specify the order of initializers, or their priority, and then sort according to it. But, currently, I don't see any value in doing it, unless somebody asks for it.

Reply:
> I don't see any value in doing it, unless somebody asks for it.

There is actually an issue for this open: https://github.com/jakartaee/servlet/issues/548

> If the order is initially random

It seems initially random indeed, but the same order between most restarts (as fas as I could see). Only occasionally WebSocket comes after Faces. For GlassFish 8 we could maybe look at a fixed/known order.

Reply:
> Then it could be enough to move the Tyrus initializer to the beginning of the list. The code would be less complicated and would also solve this issue. If the order is random now, then we can change the order in any way, without breaking anything, right? :)

Or move Faces initializer at end of the list, even without sorting. This happens in current PR when Tyrus initializer not present.

Reply:
I was also thinking if we can't devise (for the default initializers) a logical order, which we may eventually propose to standardise in EE. E.g. Servlet first, then CDI, then WebSocket, then Faces, etc...